design/adding_anova_to_mmrm_design_document.qmd

```diff
@@ -0,0 +1,579 @@
+
+If the user provides only one model fit to `anova.mmrm()`, the function will calculate and return the significance of the model terms in a data frame.
+
+#### Comparing multiple model fits
+
+If the user provides more than one model fit, a data frame is produced with a row for each model, and the columns will contain identifying information as well as standard diagnostics (e.g., AIC; log likelihood).
+
+If the user supplied `test = TRUE`, the model fits are subject to further tests to ensure they are nestable. If the data sets for the fits are not the same and the user supplied `refit = TRUE`, the data sets are harmonized and the models are refitted using the harmonized data set. The pairs of models then undergo likelihood ratio testing, and the results are augmented into the data frame of results.
+
+### Prototype code
+
+#### `anova.mmrm()`
+
+```{r }
+anova.mmrm <- function (object, ..., test = TRUE, refit = FALSE) {
+  checkmate::assert_class(object, "mmrm")
+  
+  # If there are objects supplied to "..."
+  if (...length()) {
+    
+    # Gather all objects into one list
+    fits <- list(object, ...)
+    
+    # Ensure all objects in ... are mmrm fits
+    lapply(fits[-1L], checkmate::assert_class, classes = "mmrm", .var.name = "...")
+    
+    if (refit) {
+      fits <- h_fits_data_identical(fits, refit = TRUE)
+    }
+  
+    LL_vec <- lapply(fits, logLik.mmrm_tmb)
+    fit_calls <- lapply(fits, component, "call")
+    
+    # Calculate the standard diagnostics.
+    out <-
+      data.frame(
+        call = vapply(fit_calls, deparse1, character(1L)),
+        Model = seq_along(fits),
+        df = vapply(LL_vec, attr, FUN.VALUE = numeric(1L), "df"), 
+        AIC = vapply(fits, AIC, numeric(1)),
+        BIC = vapply(fits, BIC, numeric(1)),
+        logLik = as.numeric(LL_vec)
+      )
+    
+    # If the user has requested the likelihood ratio test...
+    if (test) {
+      # First iterate through and check the models to make sure their covariates
+      # and covariance structures are nested
+      for (i in seq_along(fits)[-1L]) {
+        h_assert_nested_models(fits[[i - 1L]], fits[[i]])
+      }
+      
+      if (!refit && !h_fits_data_identical(fits, refit = FALSE)) {
+        stop("Not all data sets underlying the model fits are identical and ",
+             "refit = FALSE.", call. = FALSE)
+      }
+      
+      # Label for the pair being compared (e.g., "3 vs 4")
+      out$test <- c(NA, paste(head(out$Model, -1), "vs", tail(out$Model, -1)))
+      
+      # Calculate the likelihood ratio between the pairs using base::diff()
+      out$likelihood_ratio <- c(NA, diff(out$logLik))
+      
+      # Calculate the p-value using the ratios and differences in df
+      out$p_value <-
+        pchisq(
+          2 * abs(out$likelihood_ratio),
+          df = abs(diff(out$df)), 
+          lower.tail = FALSE
+        )
+    }
+    
+  } else { 
+    
+    # attributes of terms.object for the model formula without the cov struct
```

Comment:
the term wise testing probably makes sense to factor out in a small dedicated helper function